modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentResourceImpl.java

```diff
@@ -689,6 +691,18 @@ private ServiceContext _createServiceContext(
 		return serviceContext;
 	}
 
+	private long[] _getAncestorSiteAndDepotGroupIds(long groupId) {
+		try {
+			return _siteConnectedGroupGroupProvider.
+				getAncestorSiteAndDepotGroupIds(groupId, true);
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException);
+
+			return new long[0];
+		}
+	}
+
```

Comment:
Q: just a question, what failure are you catching here? Wouldn't it be better to let it fail?

Reply:
I was catching a PortalExecption thrown by _siteConnectedGroupGroupProvider.getAncestorSiteAndDepotGroupIds however, you're right, it should be thrown and handled later.